server/src/bundle-validation.ts

```diff
@@ -81,6 +82,16 @@ export class BundleValidationQueue {
   }
 }
 
+export const collectionDiagnostic = async (): Promise<Diagnostic[] | null> => {
```

Comment:
Minor: Could the return type here just be `Promise<Diagnostic[]>` since I believe it's definitely going to return a non-null value?